File: sdk/ruby/lib/telecom_sdk/system_api.rb
module TelecomSDK
  # API for system management
  class SystemAPI
    def initialize(client)
      @client = client
    end

    def get_stats
      @client.get("/v1/system/stats")
    end

    def get_health
      @client.get("/v1/health")
    end
  end
end

File: sdk/elixir/lib/telecom_sdk/system_api.ex
defmodule TelecomSDK.SystemAPI do
  @moduledoc """
  API for system management
  """

  defstruct [:client]

  def new(client) do
    %__MODULE__{client: client}
  end

  def get_stats(api) do
    TelecomSDK.HTTPClient.get(api.client, "/v1/system/stats")
  end

  def get_health(api) do
    TelecomSDK.HTTPClient.get(api.client, "/v1/health")
  end
end
